### DISCUSSION TITLE
Write Permission Error

### DISCUSSION THREAD

**Discussion by AtroposOrbis** [mbround18/valheim-docker#66] [Q&A] [answered]
> @AtroposOrbis This looks like your user id and group id do not match the dockers user & group id. Are you on windows? or linux?
> 
> If on linux when you run `id -u` the number it spits out should be the number you supply as an environment variable called `PUID`

When I updated the compose file and attempted another run, the same error occurred...
![image](https://user-images.githubusercontent.com/8618455/107672151-9a860000-6c62-11eb-828f-27b1c06e01c9.png)

![image](https://user-images.githubusercontent.com/8618455/107671983-67dc0780-6c62-11eb-930a-be8295df67c2.png)

_Originally posted by @AtroposOrbis in https://github.com/mbround18/valheim-docker/discussions/28#discussioncomment-360240_

**Comment by mbround18** [comment]
On your host what are the permissions of ./valheim/saves and ./valheim/server ? also as per my reply whats your user and group id? if they are not 1000 you need to supply them as environment variables.

**Comment by AtroposOrbis** [comment]
So those are required now? In the documentation here (below) it says that they are not required.
![image](https://user-images.githubusercontent.com/8618455/107672777-4deef480-6c63-11eb-8a5c-60f3a9767896.png)

Should I create the saves & server directories prior to running `docker-compose up`?

**Comment by mbround18** [comment]
I should be more clear in the documentation, they are required if your user and group ID are not 1000.

**Comment by AtroposOrbis** [comment]
@mbround18 I supplied the ID of 0/0 as I'm logged in as root. Alas, the error still occurs. Status code 8 still.

**Comment by mbround18** [comment]
might I suggest you create a non root user, add them to the docker group, then run the docker compose as that user? It is a massive security risk to run as root.

**Comment by AtroposOrbis** [comment]
> might I suggest you create a non root user, add them to the docker group, then run the docker compose as that user? It is a massive security risk to run as root.

This is just a quick VPS so I wasn't looking to create something secure right now. Just to get the server running to verify that this container works. Does it require this step then? If so I'll try creating a user now....

**Comment by mbround18** [comment]
Yesss!!! Lol this container is designed to be run on a non root user :P that is best practice when it comes to docker and I don't just highly recommend it, it is needed! 💯

**Comment by AtroposOrbis** [comment]
T_T Alright. I'll get the commands in place and then post them here for you to hopefully add to the guide for others.

**Comment by AtroposOrbis** [comment]
@mbround18 still getting the same error... :(
![image](https://user-images.githubusercontent.com/8618455/107676903-a6c08c00-6c67-11eb-9e8f-be8940d82b21.png)

**Comment by mbround18** [comment]
Is this steam user in the docker group? (In order to spin up the compose) 

Sometimes docker compose has funky permissions and it's dependant on how you set it up. You can try, as the steam user, creating those directories on your host.

**Comment by AtroposOrbis** [comment, marked as answer]
> Is this steam user in the docker group? (In order to spin up the compose)
> 
> Sometimes docker compose has funky permissions and it's dependant on how you set it up. You can try, as the steam user, creating those directories on your host.

I did docker and then full perms and still no luck. I'll just going to nuke the vds and start from scratch today. -_-

**Comment by mbround18** [comment]
Best of luck! Let me know how it goes

**Reply by AtroposOrbis** [reply]
Issue was solved via a hard reinstall of the VDS. Not sure how I messed it up. New issue has appeared however so I'll start a discussion related to that instead.
